### PR TITLE
Spotify player

### PR DESCRIPTION
**Pull request by sixatsick**
I don't know how to merge my files into your repository, I will try it anyway.

The player displays the current song and has play/pause, previous and next buttons.

### CODE REVIEW

**Comment by x70b1** [comment]
Thanks! 
I hope I can check this in the next days and give you some feedback.

File: polybar-scripts/player-spotify/player-spotify.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
```
Comment (x70b1) [inline polybar-scripts/player-spotify/player-spotify.sh]: At first please solve all shellcheck issues.

Comment (sixatsick) [reply polybar-scripts/player-spotify/player-spotify.sh]: What issues are you referring to?

Comment (132ikl) [reply polybar-scripts/player-spotify/player-spotify.sh]: https://www.shellcheck.net/ probably

File: polybar-scripts/player-spotify/player-spotify.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  artist=$(echo "$meta" | sed -nr '/xesam:artist"/,+2s/^ +string "(.*)"$/\1/p' | tail -1  | sed 's/\&/\\&/g' | sed 's#\/#\\/#g')
+  album=$(echo "$meta" | sed -nr '/xesam:album"/,+2s/^ +variant +string "(.*)"$/\1/p' | tail -1| sed 's/\&/\\&/g'| sed 's#\/#\\/#g')
+  title=$(echo "$meta" | sed -nr '/xesam:title"/,+2s/^ +variant +string "(.*)"$/\1/p' | tail -1 | sed 's/\&/\\&/g'| sed 's#\/#\\/#g')
+
+  echo "${*:-%artist% - %title%}" | sed "s/%artist%/$artist/g;s/%title%/$title/g;s/%album%/$album/g"i | sed "s#\\&#\\&#g" | sed "s#\\/#\\/#g"
+
+}
+
+main "$@"
```
Comment (x70b1) [inline polybar-scripts/player-spotify/player-spotify.sh]: Is there a reason why you put everything in a `main` function?

File: polybar-scripts/player-spotify/player-spotify.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+main() {
+  if ! pgrep -x spotify >/dev/null; then
+    echo ""; exit
+  fi
```
Comment (x70b1) [inline polybar-scripts/player-spotify/player-spotify.sh]: Please use an if-else structure. Exiting somewhere in the code is a bit unclean.

File: polybar-scripts/player-spotify/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+- dbus
+
+
+## Module
+
+```ini
+[module/spotify]
+type= custom/script
+exec = ~/.config/polybar/spotify.sh
+label =/ %{A1:dbus-send --print-reply --dest=org.mpris.MediaPlayer2.spotify /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player.Previous:}<%{A} %{A1:dbus-send --print-reply --dest=org.mpris.MediaPlayer2.spotify /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player.PlayPause:} %output% %{A} %{A1:dbus-send --print-reply --dest=org.mpris.MediaPlayer2.spotify /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player.Next:}>%{A}
```
Comment (x70b1) [inline polybar-scripts/player-spotify/README.md]: We have a lot of code here in our config file. An idea how to reduce this?
You could use functions in your script and call the different functions with a script parameter.

File: polybar-scripts/player-spotify/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+format = <label>
+tail = true
```
Comment (x70b1) [inline polybar-scripts/player-spotify/README.md]: Something is wrong here. 
Does the script create permanent output? Then use `tail = true` else use `interval = `.

**Pull request closed without merging**